client/state/data-getters/index.js

```diff
@@ -209,3 +209,62 @@ export const requestSiteAlerts = siteId => {
 		}
 	);
 };
+
+export const requestAtomicSFTPDetails = siteId =>
+	requestHttpData(
+		`atomic-hosting-data-${ siteId }`,
+		http(
+			{
+				method: 'GET',
+				path: `/sites/${ siteId }/hosting/ssh`,
+				apiNamespace: 'wpcom/v2',
+			},
+			{}
+		),
+		{
+			freshness: 5 * 60 * 1000,
+			fromApi: () => ( { username } ) => {
+				return [ `atomic-hosting-data-${ siteId }`, { username } ];
+			},
+		}
+	);
+
+export const resetAtomicSFTPUserPassword = siteId =>
+	requestHttpData(
+		`atomic-hosting-data-${ siteId }`,
+		http(
+			{
+				method: 'POST',
+				path: `/sites/${ siteId }/hosting/ssh/reset-password`,
+				apiNamespace: 'wpcom/v2',
+				body: {},
+			},
+			{}
+		),
+		{
+			fromApi: () => ( { username, password } ) => {
+				return [ `atomic-hosting-data-${ siteId }`, { username, password } ];
+			},
+			freshness: 0,
+		}
+	);
+
+export const createAtomicSFTPUser = siteId =>
+	requestHttpData(
+		`atomic-hosting-data-${ siteId }`,
+		http(
+			{
+				method: 'POST',
+				path: `/sites/${ siteId }/hosting/ssh`,
+				apiNamespace: 'wpcom/v2',
+				body: {},
```

Comment:
aha - that is probably why the POST calls would never work for me - I didn't include a body property!

Reply:
it took me a sec! it doesn't fail gracefully, ahah, and reverts to calling a 'GET' on a v1.1 version of the url if the body isn't there 🤦‍♂

Reply:
this sounds painful and unexpected - I'm going to earmark this to do a better job informing you why it doesn't work as you expected. right now it looks like it passes the `body` on downward to `wpcom.js` where I suspect the failure happens.

@glendaviesnz do you see an error in the console when this happens?

Reply:
No, the only console error was the one caused by the 404 due to the invalid endpoint.